src/OpenSage.Game/Terrain/HeightMap.cs

```diff
@@ -27,10 +27,17 @@ public float GetHeight(float x, float y)
             x = x / HorizontalScale + _heightMapData.BorderWidth;
             y = y / HorizontalScale + _heightMapData.BorderWidth;
 
-            if (x >= Width || y >= Height || x < 0 || y < 0)
-            {
-                return 0;
-            }
+            if (x < 0)
+                x = 0;
+
+            if (x >= Width)
```

Comment:
I think `Math.Clamp` would make this more readable: https://docs.microsoft.com/de-de/dotnet/api/system.math.clamp?view=netcore-3.0
e.g. `x = Math.Clamp(x, 0, Width -1);`